src/overlays/actors/ovl_Bg_Ice_Turara/z_bg_ice_turara.c

```diff
@@ -27,23 +41,164 @@ const ActorInit Bg_Ice_Turara_InitVars = {
     (ActorFunc)BgIceTurara_Update,
     (ActorFunc)BgIceTurara_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ice_Turara/BgIceTurara_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ice_Turara/BgIceTurara_Destroy.s")
+InitChainEntry sInitChain[] = {
+    ICHAIN_F32(uncullZoneScale, 600, ICHAIN_CONTINUE),
+    ICHAIN_F32(gravity, -3, ICHAIN_CONTINUE),
+    ICHAIN_F32(minVelocityY, -30, ICHAIN_CONTINUE),
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
+};
+
+// gravitational acceleration
+Vec3f D_8089267C[] = { 0.0f, -1.0f, 0.0f };
+
+Color_RGBA8 primColor[] = { 170, 255, 255, 255 };
+
+Color_RGBA8 envColor[] = { 0, 50, 100, 255 };
```

Comment:
A few things:

- Do these have to be arrays?
- You can name `D_8089267C` "gravity" if you're sure that's what it is.
- Since these are only used in `BgIceTurara_Break`, they can be declared in the body of that function.

Reply:
that vector is just `accel` as noted by the protoype for the effect spawn function. In most cases you can just name it after what its called there.
and +1 on moving this data into the function. youll need to make it static inside the function though.
and no they dont have to be arrays

Reply:
I have moved them inside the function as statics. I have seen some declared elsewhere as nonstatic; is this just due to how the original data was structured?

Reply:
For inline declarations (that is, declarations inside the function and not declared as static), the code will copy the data onto the stack. For structs this isn't any different from declaring it as static and then struct copying it into a stack variable, but for arrays it must be inlined.